File: ThreePalindrome.java
import java.util.*;

public class ThreePalindrome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();
        sc.close();
        int l = s.length();

        for (int i = 1; i < l - 1; i++) {
            String s1 = s.substring(0, i);
            if (isPalindrome(s1)) {
                for (int j = 1; i + j < l; j++) {
                    String s2 = s.substring(i, i + j);
                    String s3 = s.substring(i + j);
                    if (isPalindrome(s2) && isPalindrome(s3)) {
                        System.out.println(s1);
                        System.out.println(s2);
                        System.out.println(s3);
                        System.exit(0);
                    }
                }
            }
        }

        System.out.println("Impossible");
    }

    private static boolean isPalindrome(String s) {
        if (s.length() == 1) {
            return true;
        }
        String s1 = new StringBuilder(s).reverse().toString();
        return s1.equals(s);
    }
}

File: ThreePalindrome/ThreePalindrome.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

bool is_palindrome(string s) {
  if (s.length() == 1)
    return true;
  string s1 = s;
  reverse(s.begin(), s.end());
  return s1 == s;
}

int main() {
  string s, s1, s2, s3;
  cin >> s;
  int l = s.length();
  for (int i = 1; i < l - 1; i++) {
    s1 = s.substr(0, i);
    if (is_palindrome(s1)) {
      for (int j = 1; j < l - i; j++) {
        s2 = s.substr(i, j);
        s3 = s.substr(i + j, l - i - j);
        if (is_palindrome(s1) && is_palindrome(s2) && is_palindrome(s3)) {
          cout << s1 << endl << s2 << endl << s3 << endl;
          return 0;
        }
      }
    }
  }
  cout << "Impossible" << endl;
  return 0;
}